evm_arithmetization/src/generation/jumpdest.rs

```diff
@@ -0,0 +1,114 @@
+use std::{
+    collections::{BTreeSet, HashMap},
+    fmt::Display,
+    ops::{Deref, DerefMut},
+};
+
+use keccak_hash::H256;
+use serde::{Deserialize, Serialize};
+
+/// Each `CodeAddress` can be called one or more times, each time creating a new
+/// `Context`. Each `Context` will contain one or more offsets of `JUMPDEST`.
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize, Default)]
+pub struct ContextJumpDests(pub HashMap<usize, BTreeSet<usize>>);
+
+/// The result after proving a `JumpDestTableWitness`.
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize, Default)]
+pub(crate) struct JumpDestTableProcessed(pub HashMap<usize, Vec<usize>>);
```

Comment:
When I was diving into this, it made it more clear what the purpose of both types was.  But otherwise it provides a bit of encapsulation and type safety.